examples/basic_infection/README.md

```diff
@@ -0,0 +1,103 @@
+
+Infected individuals schedule their recovery at time `t + infected period`. The infection status of recovered individuals remains as recovered for the rest of the simulation.
+
+### Infection state transitions
+```mermaid
+flowchart LR
+S(Susceptible) --FoI--> I(Infected) --inf. period--> R(Recovered)
+```
+## Simulation architecture
+The simulation executaion first initializes the model parameters and prepares the simulation modules.
+
+### Initialization and modules
+
+ - *parameters module*: this module manages data for parameter values for population size, force of infection, and infection period. At initialization, all parameter variables are set to a specific value.
+ - *person module*: contains a unique ID for each person that identifies each person in the simulation.
+ - *`person_infection_status` module*: this module connects each person ID with a specific value for a person's infection status, which could be one of Susceptible, Infected, or Recovered.
+ - *population manager module*: The population manager module is in charge of setting up the population. At initialization, the module reads the population size (N) parameter and creates N persons with unique ids (1..N) and sets their initial infection status to Susceptible.
+ - *transmission module*: This module is in charge of spreading the infection through the population. At initialization, it schedules an infection attempt for time 0. An infection attempt consists of choosing at random a susceptible individual from the population and setting their infection status to **infected**. Each infection attempt, also schedules the next infection attempt, which time is based on the force of infection.
+ - *infection manager module*: This module controls the infection status of each person. Transmission of pathogen is not handled in this module, only individual's infection status. Specifically, it handles the progression of newly infected persons by scheduling recovery, which is drawn from an exponential distribution based on the infection period parameter from the parameters module.
+ - *reports module*: This module reports changes in infection status.
+ - *random_number_generator module*: Two random number generators are required. One to control the sequence of susceptible persons drawn in the transmission module, and another to control the specific times of recovery in the infection manager module.
+
+### Control flow
+#### Trasmission manager
+- Module data: This modules doesn't hold any specific data
+- Dependencies:
+```
+    context, parameters, random number generator, person infection status, person, population manager
+```
+- Initialization:
+```
+    init(context) {
+        context.add_rng(id = transmission);
+        context.add_plan(attempt_infection(context), time = 0);
```

Comment:
Ah, so basically the first thing that happens is you try to infect someone?

Reply:
Yes. Although, not sure that I put it in the best place. In this example, there isn't any transmission through contacts, but here it assumes that everything else has been setup, meaning the population has been set properly with the infection status as susceptible.